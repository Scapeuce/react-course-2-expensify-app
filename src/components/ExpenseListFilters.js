import React from 'react'
import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'
import  {DateRangePicker} from  'react-dates'

class ExpenseListFilters extends React.Component {
    state = {
    calendarFocused: null
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate))
        this.props.dispatch(setEndDate(endDate))
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}))
    }
    onTextChange = (e) => {
                    this.props.dispatch(setTextFilter(e.target.value))
            
    }
    onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  };
    render() {
        return (
            <div>
                <input
                    type="text"
                    value={this.props.filters.text}
                    onChange={this.onTextChange} />
                <select value={this.props.filters.sortBy}
                    onChange={this.onSortChange}>
                    <option value='amount'>Amount</option>
                    <option value='date'>Date</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    startDateId='ST_DATE'
                    endDateId='END_DATE'
                    numberOfMonths={1}
                    isOutsideRange={()=>false}

                />
            
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}
export default connect(mapStateToProps) (ExpenseListFilters);