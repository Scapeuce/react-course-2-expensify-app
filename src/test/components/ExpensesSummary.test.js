import React from 'react'
import { shallow } from 'enzyme'
import {ExpensesSummary}  from '../../components/ExpensesSummary'
// import expensesTotal from '../../selectors/expenses-total';
test('shoud correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});
test('shoud correctly render ExpensesSummary with  more than 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={236677542319} />);
    expect(wrapper).toMatchSnapshot();
});