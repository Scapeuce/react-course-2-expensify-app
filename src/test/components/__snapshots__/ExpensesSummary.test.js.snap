// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud correctly render ExpensesSummary with  more than 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $2,366,775,423.19
  </h1>
</div>
`;

exports[`shoud correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;
